File: OPENAPI-REST-API/openapi-client/scalaz/src/main/scala/org/openapitools/client/api/QuickserverRow.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import QuickserverRow._

case class QuickserverRow (
  /* The id of the qs. */
  qs_id: String,
/* The name of the qs. */
  qs_name: String,
/* The cost of the qs. */
  cost: String,
/* The hostname of the qs. */
  qs_hostname: String,
/* The status of the qs. */
  qs_status: String,
/* The comment of the qs. */
  qs_comment: String)

object QuickserverRow {
  import DateTimeCodecs._

  implicit val QuickserverRowCodecJson: CodecJson[QuickserverRow] = CodecJson.derive[QuickserverRow]
  implicit val QuickserverRowDecoder: EntityDecoder[QuickserverRow] = jsonOf[QuickserverRow]
  implicit val QuickserverRowEncoder: EntityEncoder[QuickserverRow] = jsonEncoderOf[QuickserverRow]
}

File: OPENAPI-REST-API/openapi-client/zapier/models/QuickserverRow.js
const utils = require('../utils/utils');

module.exports = {
    fields: (prefix = '', isInput = true, isArrayChild = false) => {
        const {keyPrefix, labelPrefix} = utils.buildKeyAndLabel(prefix, isInput, isArrayChild)
        return [
            {
                key: `${keyPrefix}qs_id`,
                label: `The id of the qs. - [${labelPrefix}qs_id]`,
                required: true,
                type: 'string',
            },
            {
                key: `${keyPrefix}qs_name`,
                label: `The name of the qs. - [${labelPrefix}qs_name]`,
                required: true,
                type: 'string',
            },
            {
                key: `${keyPrefix}cost`,
                label: `The cost of the qs. - [${labelPrefix}cost]`,
                required: true,
                type: 'string',
            },
            {
                key: `${keyPrefix}qs_hostname`,
                label: `The hostname of the qs. - [${labelPrefix}qs_hostname]`,
                required: true,
                type: 'string',
            },
            {
                key: `${keyPrefix}qs_status`,
                label: `The status of the qs. - [${labelPrefix}qs_status]`,
                required: true,
                type: 'string',
            },
            {
                key: `${keyPrefix}qs_comment`,
                label: `The comment of the qs. - [${labelPrefix}qs_comment]`,
                required: true,
                type: 'string',
            },
        ]
    },
    mapping: (bundle, prefix = '') => {
        const {keyPrefix} = utils.buildKeyAndLabel(prefix)
        return {
            'qs_id': bundle.inputData?.[`${keyPrefix}qs_id`],
            'qs_name': bundle.inputData?.[`${keyPrefix}qs_name`],
            'cost': bundle.inputData?.[`${keyPrefix}cost`],
            'qs_hostname': bundle.inputData?.[`${keyPrefix}qs_hostname`],
            'qs_status': bundle.inputData?.[`${keyPrefix}qs_status`],
            'qs_comment': bundle.inputData?.[`${keyPrefix}qs_comment`],
        }
    },
}
